src/lib/progress.h

```diff
@@ -0,0 +1,65 @@
+#ifndef __PROGRESS__
+#define __PROGRESS__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct step {
+	unsigned int current;
+	unsigned int total;
+	char *description;
+};
+
+/*
+ * available formats the output can be printed
+ */
+enum progress_formats {
```

Comment:
it would be better to use the same approach on log for that. Instead of having one enum for types we can have a function pointer for a report_progress(). We can have a type set to start, end or progress. And the json could print the start/end messages or progress messages depending on the type.

Reply:
ok, good idea